File: byeongmin/이분탐색/암기왕.java
package binarySearch;

import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

public class 암기왕 {
    public static boolean binarySearch(int[] arr, int target, int start, int end) {
        if (start > end)
            return false;

        int mid = (start + end) / 2;

        if (arr[mid] == target)
            return true;
        else if (arr[mid] < target)
            return binarySearch(arr, target, mid + 1, end);
        else
            return binarySearch(arr, target, start, mid - 1);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            int n = Integer.parseInt(br.readLine());
            int[] arr = new int[n];
            StringTokenizer st = new StringTokenizer(br.readLine());
            for (int j = 0; j < n; j++) {
                arr[j] = Integer.parseInt(st.nextToken());
            }
            Arrays.sort(arr);

            int m = Integer.parseInt(br.readLine());

            st = new StringTokenizer(br.readLine());

            for (int j = 0; j < m; j++) {
                int target = Integer.parseInt(st.nextToken());
                if (binarySearch(arr, target, 0, n - 1))
                    bw.write(1 + "\n");
                else
                    bw.write(0 + "\n");
            }
        }
        bw.close();
    }
}

File: minji/이분탐색/암기왕.py
import sys


def binary_search(array, target, start, end):
    while start <= end:
        mid = (start + end) // 2
        if array[mid] == target:
            return mid
        elif array[mid] > target:
            end = mid - 1
        else:
            start = mid + 1
    return None


t = int(sys.stdin.readline().rstrip())
for _ in range(t):
    n = int(sys.stdin.readline().rstrip())
    array1 = list(map(int, sys.stdin.readline().split()))
    array1.sort()
    m = int(sys.stdin.readline().rstrip())
    array2 = list(map(int, sys.stdin.readline().split()))

    for i in array2:
        result = binary_search(array1, i, 0, n - 1)
        if result is not None:
            print(1)
        else:
            print(0)